src/test/java/be/seeseemelk/mockbukkit/inventory/meta/BookMetaMockTest.java

```diff
@@ -0,0 +1,303 @@
+	{
+		meta.setGeneration(BookMetaMock.Generation.ORIGINAL);
+		assertEquals(BookMetaMock.Generation.ORIGINAL, meta.getGeneration());
+	}
+
+	@Test
+	void test_hashCode_SameObject()
+	{
+		assertEquals(meta.hashCode(), meta.hashCode());
+	}
+
+	@Test
+	void test_hashCode_DifferentObject_SameTitle()
+	{
+		BookMetaMock meta1 = new BookMetaMock();
+		BookMetaMock meta2 = new BookMetaMock();
+		assertEquals(meta1.hashCode(), meta2.hashCode());
+
+		meta1.setTitle("Bumba");
+		meta2.setTitle("Bumba");
+		assertEquals(meta1.hashCode(), meta2.hashCode());
+	}
+
+	@Test
+	void test_hashCode_DifferentObject_DifferentTitle()
+	{
+		BookMetaMock meta1 = new BookMetaMock();
+		meta1.setTitle("Bumba1");
+
+		BookMetaMock meta2 = new BookMetaMock();
+		meta2.setTitle("Bumba2");
+
+		assertNotEquals(meta1.hashCode(), meta2.hashCode());
+	}
+
+	@Test
+	void test_equals_SameObject()
+	{
+        assertEquals(meta, meta);
+	}
+
+	@Test
+	void test_equals_DifferentObject()
+	{
+        assertNotEquals(meta, Material.DIAMOND);
+
+		BookMetaMock meta2 = meta.clone();
+		meta2.setUnbreakable(!meta.isUnbreakable()); // This is not from BookMeta, but from ItemMeta
+		assertNotEquals(meta, meta2);
+	}
+
+	@Test
+	void test_equals_SameAttributes()
+	{
+		BookMetaMock meta1 = new BookMetaMock();
+		BookMetaMock meta2 = new BookMetaMock();
+
+		meta1.setTitle("Bumba");
+		meta1.setAuthor("De Lu");
+		meta1.addPage("Page1");
+		meta2.setTitle("Bumba");
+		meta2.setAuthor("De Lu");
+		meta2.addPage("Page1");
+
+        assertEquals(meta1, meta2);
+	}
+
+	@Test
+	void test_equals_DifferentTitle()
+	{
+		BookMetaMock cloned = meta.clone();
+
+		meta.setTitle("Bumba1");
+		cloned.setTitle("Bumba2");
+
+        assertNotEquals(meta, cloned);
+	}
+
+	@Test
+	void test_equals_DifferentAuthor()
+	{
+		BookMetaMock cloned = meta.clone();
+
+		meta.setAuthor("Bumba1");
+		cloned.setAuthor("Bumba2");
+
+		assertNotEquals(meta, cloned);
+	}
+
+	@Test
+	void test_equals_DifferentPage()
+	{
+		BookMetaMock cloned = meta.clone();
+
+		meta.addPage("Bumba1");
+		cloned.addPage("Bumba2");
+
+		assertNotEquals(meta, cloned);
+	}
+
+	@Test
+	void test_equals_DifferentPages()
+	{
+		BookMetaMock cloned = meta.clone();
+
+		meta.addPage("Bumba1", "Bumba3");
+		cloned.addPage("Bumba1");
+
+		assertNotEquals(meta, cloned);
+	}
+
+	@Test
+	void test_hasTitle()
+	{
+		assertFalse(meta.hasTitle());
+		meta.setTitle("Bumba");
+		assertTrue(meta.hasTitle());
+	}
+
+	@Test
+	void test_hasAuthor()
+	{
+		assertFalse(meta.hasAuthor());
+		meta.setAuthor("Bumba");
+		assertTrue(meta.hasAuthor());
+	}
+
+	@Test
+	void test_getPages()
+	{
+		assertTrue(meta.getPages().isEmpty());
+
+		List<String> pages = new ArrayList<String>();
+		pages.add("Page1");
+		pages.add("Page2");
+
+		meta.setPages(pages);
+		assertEquals(pages, meta.getPages());
+
+		pages.set(0, "Dummy");
+		assertNotEquals(pages, meta.getPages()); // testing that the `pages`' array modifications is not reflected in the book.
```

Comment:
It'd be clearer here to just do `assertNotSame(pages, meta.getPages())` which performs a reference equality check.

Reply:
Ok, didn't know that one.

I changed it now to:
```java
		meta.setPages(pages);
		assertNotSame(pages, meta.getPages());
		assertEquals(pages, meta.getPages());

		pages.set(0, "Dummy");
		assertNotSame(pages, meta.getPages()); // testing that the `pages`' array modifications is not reflected in the book.
		assertNotEquals(pages, meta.getPages()); // testing that the `pages`' array modifications is not reflected in the book.
```

ie: checking that they are NOT the same, but ARE equal.
And secondly, that they're still not the same, but now they're not equal too.